src/test/java/org/terasology/replayTests/FallDamageTest.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2018 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.replayTests;
+
+import org.junit.After;
+import org.junit.Test;
+import org.terasology.ReplayTestingEnvironment;
+import org.terasology.TestUtils;
+import org.terasology.engine.GameThread;
+import org.terasology.entitySystem.entity.EntityRef;
+import org.terasology.logic.health.HealthComponent;
+import org.terasology.logic.players.LocalPlayer;
+import org.terasology.recording.RecordAndReplayStatus;
+import org.terasology.registry.CoreRegistry;
+
+import static org.junit.Assert.assertNotEquals;
+
+public class FallDamageTest {
+    private ReplayTestingEnvironment environment = new ReplayTestingEnvironment();
+
+    private Thread replayThread = new Thread() {
+
+        @Override
+        public void run() {
+            try {
+                String replayTitle = "FallDamage";
+                environment.openReplay(replayTitle, true);
+            } catch (Exception e) {
+                throw new RuntimeException(e);
+            }
+        }
+    };
+
+    @After
+    public void closeReplay() throws Exception {
+        environment.getHost().shutdown();
+        GameThread.reset();
+        replayThread.join();
+    }
+
+    @Test
+    public void testFallDamage() {
+
+        replayThread.start();
+
+        LocalPlayer localPlayer = CoreRegistry.get(LocalPlayer.class);
+        TestUtils.waitUntil(() -> localPlayer.isValid());
```

Comment:
You should wait for the environment to be initialized first. So the right order is:

```java
TestUtils.waitUntil(() -> (environment.isInitialised() && environment.getRecordAndReplayStatus() == RecordAndReplayStatus.REPLAYING));
TestUtils.waitUntil(() -> localPlayer.isValid());
```

Without this change I'm getting the following error message:

```java
23:27:57.860 [main] ERROR org.terasology.TestUtils - Error has occurred in the waitUntil method: 
java.lang.NullPointerException: null
	at org.terasology.replayTests.FallDamageTest.lambda$testFallDamage$0(FallDamageTest.java:60)
	at org.terasology.TestUtils.waitUntil(TestUtils.java:32)
	at org.terasology.replayTests.FallDamageTest.testFallDamage(FallDamageTest.java:60)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:68)
	at com.intellij.rt.execution.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:51)
	at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:242)
	at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:70)
```

Reply:
The change has been made!